sympy/geometry/tests/test_polygon.py

```diff
@@ -611,3 +613,44 @@ def test_cut_section():
     p3, p4 = p.cut_section(Line((0, 0), slope=0))
     assert p3 == Polygon(Point2D(0, 10), Point2D(0, 0), Point2D(20, 0), Point2D(20, 10))
     assert p4 == None
+
+    # case where the line does not intersect with a polygon at all
+    raises(ValueError, lambda: p.cut_section(Line((0, a), slope=0)))
+
+def test_type_of_triangle():
+    # Isoceles triangle
+    p1 = Polygon(Point(0,0), Point(5,0), Point(2,4))
+    assert p1.is_isosceles() == True
+    assert p1.is_scalene() == False
+    assert p1.is_equilateral() == False
+
+    # Scalene triangle
+    p2 = Polygon (Point(0,0), Point(0,2), Point(4,0))
+    assert p2.is_isosceles() == False
+    assert p2.is_scalene() == True
+    assert p2.is_equilateral() == False
+
+    # Equilateral triagle
+    p3 = Polygon(Point(0,0), Point(6,0), Point(3, sqrt(27)))
+    assert p3.is_isosceles() == True
+    assert p3.is_scalene() == False
+    assert p3.is_equilateral() == True
+
+def test_do_poly_distance():
+    # Non-intersecting polygons
+    square1 = Polygon (Point(0,0), Point(0,1), Point(1,1), Point(1,0))
+    triangle1 = Polygon(Point(1, 2), Point(2, 2), Point(2, 1))
+    assert square1._do_poly_distance(triangle1) == sqrt(2)/2
+
+    # Polygons which sides intersect
+    square2 = Polygon(Point(1,0), Point(2,0), Point(2,1), Point(1,1))
+    with warns(UserWarning, \
+               match="Polygons may intersect producing erroneous output"):
+        assert square1._do_poly_distance(square2) == 0    
```

Comment:
```suggestion
        assert square1._do_poly_distance(square2) == 0
```

Reply:
I don't think that this had been resolved yet

Reply:
@johan1505 Please remove the white spacing at the end of line 649, as this is why it fails the ci tests.